statsmodels/stats/anova.py

```diff
@@ -362,6 +367,197 @@ def anova_lm(*args, **kwargs):
     return table
 
 
+def _not_slice(slices, slices_to_exclude, n):
+    ind = np.array([True]*n)
+    for term in slices_to_exclude:
+        s = slices[term]
+        ind[s] = False
+    return ind
+
+
+def _ssr_reduced_model(y, x, term_slices, params, keys):
+    """
+    Residual sum of squares of OLS model excluding factors in `keys`
+    Assumes x matrix is orthogonal
+
+    Parameters
+    ----------
+    y : array_like, dependent variables
+    x : array_like, independent variables
+    term_slices : a dict of slices
+        term_slices[key] is a boolean array specifies the parameters
+        associated with the factor `key`
+    params : OLS solution of y = x * params
+    keys : factors to be excluded
+
+    Returns
+    -------
+    residual sum of squares and degrees of freedom
+
+    """
+    ind = _not_slice(term_slices, keys, x.shape[1])
+    params1 = params[ind]
+    ssr = np.subtract(y, x[:, ind].dot(params1))
+    ssr = ssr.T.dot(ssr)
+    df_resid = len(y) - len(params1)
+    return ssr, df_resid
+
+
+class AnovaRM(object):
+    """
+    Repeated measures Anova using least squares regression
+
+    The full model regression residual sum of squares is
+    used to compared with reduced model for calculating the
+    within subject effect sum of squares [1]
+
+    Between subject effect and Greenhouse-Geisser correction is not yet
+    supported.
+
+    Parameters
+    ----------
+    data : DataFrame
+    dv : string
```

Comment:
use `depvar` instead of `dv` 
and we leave it otherwise as is.

It's not the same kind of pattern as the models, but this is a hypothesis test "function"